pkg/ipcache/cidr.go

```diff
@@ -140,7 +140,7 @@ func allocate(prefix *net.IPNet, lbls labels.Labels) (*identity.Identity, bool,
 
 func releaseCIDRIdentities(ctx context.Context, identities map[string]*identity.Identity) {
 	for prefix, id := range identities {
-		released, err := IdentityAllocator.Release(ctx, id)
+		released, err := IdentityAllocator.Release(ctx, id, false)
```

Comment:
So this was the bug fix, preventing recursive call to selector cache that had the lock already held.

Reply:
I wrote this in the context of the first commit, not meaning that this would have been the fix for unreleased allocations :-)

Reply:
:+1: yeah that's right.

Overall I'm not super happy with the interface here since it's weird for the identity allocator to sometimes be responsible for notifying callers and sometimes not. Ideally we would define architecturally whether it's the identity allocator subsystem's responsibility to notify users when identities are allocated. Then the code can consistently handle the notifications either above or below this interface. But that's a bigger project, for now (for backports) I wanted to keep the fix as minimal as possible.

Reply:
(It occurs to me now that the actual fix there was hidden in a large commit and I could've split the refactor vs. the fix there to make it clearer. But I guess you found it in the end.)